### DISCUSSION TITLE
What's the advantage of using tasks instead of a deployment script ?

### DISCUSSION THREAD

**Discussion by SvenMeyer** [PaulRBerg/hardhat-template#102] [Q&A]
What's the advantage of using tasks instead of a deployment script ?

**Comment by PaulRBerg** [comment]
The main argument is the nicer UX of passing arguments to the CLI command (Hardhat takes care of this when using tasks).

You can read more about tasks versus scripts in Hardhat in this thread on StackExchange:

[Where does the line blur between a task and a script in Hardhat?](https://ethereum.stackexchange.com/questions/83656/where-does-the-line-blur-between-a-task-and-a-script-in-hardhat)
